ministryofjustice/laa-apply-for-criminal-legal-aid
# CRIMAP-529 Dropzone without MOJ component

## Description of change
This is a counterpart to previous PR #460.

We may decide this is a better way, as I do believe we will need to heavily modify and customise MOJ multi-upload component anyways, so we might as well avoid using it, and instead just use a standalone dropzone module (that doesn't require jQuery) and our own styling and JS for events handling.

I've made a head start on styling and JS events preparing the structure and foundation. We can either migrate styling and JS handling from MOJ component, or from other service like Civil Apply, or Tax Tribunals, etc. that use dropzone standalone too.

Note how, out of the box, without any additional JS, dropzone handles the Ajax call, and the authenticity token is propagated, etc.

## Link to relevant ticket
https://dsdmoj.atlassian.net/browse/CRIMAP-529

## Screenshots of changes (if applicable)
<img width="625" alt="Screenshot 2023-08-16 at 08 37 50" src="https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/assets/687910/a9e547f8-f4bd-4875-8cdc-946a4d529f96">

## Notes for reviewer

## How to manually test the feature
As before, checkout, run `yarn`, run `rails assets:clobber && rails dartsass:build` and restart server. Go to the evidence upload page and the drop zone should be "functional" but it will have out of the box dropzone-supplied styling for now.
